server/utils/queryHelpers/pageGet.js

```diff
@@ -80,13 +81,41 @@ const getPubsForPageLayout = async (layout, initialData) => {
 	}
 
 	const pubsPerBlock = await Promise.all(
-		layout
+		(layout || [])
 			.filter((block) => block.type === 'pubs')
 			.map((block) => getPubsForPageBlock(block.content, initialData)),
 	);
 	return pubsPerBlock.reduce((acc, next) => [...acc, ...next], []);
 };
 
+const enrichLayoutWithPubTokens = (layout, initialData) => {
+	const { loginData, communityData } = initialData;
+	const userId = loginData && loginData.id;
+	if (layout && userId) {
+		return layout.map((block) => {
+			const { type, content } = block;
+			if (type === 'banner') {
+				const { buttonType, defaultCollectionIds } = content;
+				if (buttonType === 'create-pub') {
+					return {
+						...block,
+						content: {
+							...content,
+							createPubToken: issueCreatePubToken({
+								userId: userId,
+								communityId: communityData.id,
+								createInCollectionIds: defaultCollectionIds,
+							}),
```

Comment:
Am I right to think that if a user sits on a page for 1hr without clicking the button, then tries to click it, they will get a 403 error? If so, how is that error presented to the user?

Reply:
Well, I was going to let them flounder, but I woke up more motivated this morning and added this:

![image](https://user-images.githubusercontent.com/2208769/92268790-0775bc80-eeb1-11ea-918e-0fafce197f0b.png)